sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/services.py

```diff
@@ -15,7 +21,20 @@
 module_logger = logging.getLogger(__name__)
 
 
-class JobServiceSchema(PathAwareSchema):
+class JobServiceBaseSchema(PathAwareSchema):
+    port = fields.Int()
+    endpoint = fields.Str(dump_only=True)
+    status = fields.Str(dump_only=True)
+    nodes = fields.Str()
+    error_message = fields.Str(dump_only=True)
+    properties = fields.Dict()
+
+
+class JobServiceSchema(JobServiceBaseSchema):
+    """This is to support tansformation of job services passed as dict type and
+    internal job services like Custom, Tracking, Studio set by the system.
+    """
+
     job_service_type = UnionField(
```

Comment:
I know it will be a breaking change, it's up to you to do it or not. But to align with other polymorphic scenarios, this should be called just `type`.

Reply:
Got it. Will take it as a separate PR as this breaks all jobs including pipeline jobs.
Task 2047301: Rename JobService<>.job_service_type to JobService<>.type and re-record the tests